### PR TITLE
stub wasm put

### PR DESCRIPTION
**Pull request by neonphog**
DEPENDS ON: https://github.com/holochain/bootstrap/pull/21

- The POST/put handler in this PR isn't strictly needed, since it's not enabled, but it encapsulates the sum of my learning when spiking about feasibility and code size on including WASM in cloudflare workers that I wanted to keep track of.
  - With all the crates we need to re-implement everything (namely `rmp`, `base64`, and `ed25519-dalek`) the wasm size will be around ~140K where the minified ts is around ~180K, and workers are capped at 1M, so this is totally feasible).
- There is also some glue, largely around the KV, that will be made use of in the next follow-on PR.

### CODE REVIEW

File: rust/holochain_bootstrap_core/src/agent_info.rs [hunk 1]
```diff
@@ -0,0 +1,184 @@
+//! Agent Info Structs
+
+use crate::decode::*;
+use crate::types::*;
+
+/// Struct for decoding agent info
+pub struct AgentInfoRef<'a> {
```
Comment (thedavidmeister) [inline rust/holochain_bootstrap_core/src/agent_info.rs:7]: is it possible/desirable to use the actual kitsune types here so they can always be kept in sync with versioning?

File: rust/holochain_bootstrap_core/src/agent_info.rs [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+    pub space: &'a [u8],
+
+    /// agent id / pubkey
+    pub agent: &'a [u8],
+
+    /// urls this agent is reachable at
+    pub urls: Vec<&'a str>,
+
+    /// timestamp this blob was signed
+    pub signed_at_ms: f64,
```
Comment (thedavidmeister) [inline rust/holochain_bootstrap_core/src/agent_info.rs:18]: these are `u64` in core

File: rust/holochain_bootstrap_core/src/types.rs [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+/// Trait representing a KV implementation
+pub trait AsKV: 'static {
+    /// put data into the KV
+    fn put<'a>(&'a self, key: &str, value: &[u8], ttl_secs: f64) -> BCoreFut<'a, BCoreResult<()>>;
+
+    /// get data from the KV
+    fn get<'a>(&'a self, key: &str) -> BCoreFut<'a, BCoreResult<Box<[u8]>>>;
+
+    /// delete a key from the KV
+    fn delete<'a>(&'a self, key: &str) -> BCoreFut<'a, BCoreResult<()>>;
```
Comment (thedavidmeister) [inline rust/holochain_bootstrap_core/src/types.rs:135]: i was thinking "delete" is just update with a short expiry

File: rust/holochain_bootstrap_core/src/types.rs [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+
+    /// list keys from the KV
+    fn list<'a>(&self, prefix: Option<&str>) -> BCoreFut<'a, BCoreResult<Vec<String>>>;
```
Comment (thedavidmeister) [inline rust/holochain_bootstrap_core/src/types.rs:138]: do we want `random`? that allows to randomly get agents without doing pagination over the wire, or restricting random results that are biased to the first page only?

**Review by thedavidmeister — CHANGES_REQUESTED** [review]

File: rust/holochain_bootstrap_core/src/agent_info.rs (hunk 2 above)
Comment (neonphog) [reply rust/holochain_bootstrap_core/src/agent_info.rs]: fixed

File: rust/holochain_bootstrap_core/src/agent_info.rs (hunk 1 above)
Comment (neonphog) [reply rust/holochain_bootstrap_core/src/agent_info.rs:9]: Yeah, I was thinking kitsune would pull in these types so they'd be the same.

File: rust/holochain_bootstrap_core/src/types.rs (hunk 3 above)
Comment (neonphog) [reply rust/holochain_bootstrap_core/src/types.rs:124]: This is an FFI to the javascript, so it needs to match the cloudflare KV api.

File: rust/holochain_bootstrap_core/src/types.rs (hunk 4 above)
Comment (neonphog) [reply rust/holochain_bootstrap_core/src/types.rs:127]: This is just an FFI to the cloudflare KV api, when we get around to migrating your typescript functions into rust/wasm, I was imagining keeping your exact API, so yes, doing the random thing.

**Comment by neonphog** [comment]
@thedavidmeister - please re-review, I addressed your comments, and got the WASM size down to ~`100K` by using `#![no_std]` on the rust crates.

File: rust/holochain_bootstrap_core/src/agent_info.rs (hunk 1 above)
Comment (thedavidmeister) [reply rust/holochain_bootstrap_core/src/agent_info.rs:9]: @neonphog oh interesting, ok yeah we can do it that way

File: rust/holochain_bootstrap_core/src/types.rs (hunk 3 above)
Comment (thedavidmeister) [reply rust/holochain_bootstrap_core/src/types.rs:124]: ah i c, ok yes

**Review by thedavidmeister — APPROVED** [review]

**Pull request merged**